server/pipeline_trigger.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package server
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+func getPipelineFormData(pipelineTrigger *PipelineTrigger, args []string) url.Values {
+	data := url.Values{}
+	data.Add("token", pipelineTrigger.Token)
+	data.Add("ref", pipelineTrigger.Reference)
+	for key, value := range pipelineTrigger.Variables {
+		var v = value
+		if search := strings.TrimPrefix(value, "%%"); search != value {
```

Comment:
It was hard to figure out the pattern here until I read the unit tests. Should we document this idea of interpolating when a variable starts with `%%` somewhere?

Reply:
That is great point, i will create docs folder and give high level details about the commands.